src/components/screens/timebank/timebank-screen.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from "react";
+import { Person, Timespan } from "../../../generated/client";
+import { CircularProgress, Card } from "@mui/material";
+import { userProfileAtom } from "../../../atoms/auth";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import { errorAtom } from "../../../atoms/error";
+import {
+  dailyEntriesAtom,
+  personDailyEntryAtom,
+  personTotalTimeAtom,
+  personsAtom
+} from "../../../atoms/person";
+import { useApi } from "../../../hooks/use-api";
+import TimebankContent from "./timebank-content";
+import { DateTime } from "luxon";
+import strings from "../../../localization/strings";
+import config from "../../../app/config";
+
+/**
+ * Time bank screen component containing functions and logic to run time bank features.
+ */
+const TimebankScreen = () => {
+  const userProfile = useAtomValue(userProfileAtom);
+  const [timespanSelector, setTimespanSelector] = useState<Timespan>(Timespan.ALL_TIME);
+  const setError = useSetAtom(errorAtom);
+  const { personsApi, dailyEntriesApi } = useApi();
+  const persons = useAtomValue(personsAtom);
+  const [personTotalTimeLoading, setPersonTotalTimeLoading] = useState(false);
+  const [personTotalTime, setPersonTotalTime] = useAtom(personTotalTimeAtom);
+  const [personDailyEntry, setPersonDailyEntry] = useAtom(personDailyEntryAtom);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+
+  /**
+   * Gets the logged in person from the persons list atom and then gets that person's total time data.
+   * Function is called on initial load / persons list changes or timespan changes.
+   * @param timespan Optional Timespan string which controls whether @PersonTotalTime results are condensed into weeks, months, years or all time. Defaults to all time.
+   */
+  const getPersonTotalTime = async (timespan?: Timespan): Promise<void> => {
+    if (!personTotalTime || timespan) {
+      setPersonTotalTimeLoading(true);
+      setTimespanSelector(timespan || Timespan.ALL_TIME);
+      if (persons.length) {
+        try {
+          const loggedInPerson = persons.filter(
+            (person: Person) => person.keycloakId === config.keycloak.id || userProfile?.id
+          )[0];
+          const fetchedPersonTotalTime = await personsApi.listPersonTotalTime({
+            personId: loggedInPerson.id,
+            timespan: timespan
+          });
+          setPersonTotalTime(fetchedPersonTotalTime[0]);
+        } catch (error) {
+          setError(`${strings.error.totalTimeFetch}, ${error}`);
+        }
+        setPersonTotalTimeLoading(false);
+      }
+    }
+  };
+
+  /**
+   * Gets the logged in person from persons list atom and then fetches the person's daily entries.
+   */
+  const getPersonDailyEntries = async (): Promise<void> => {
+    if (persons.length) {
```

Comment:
Typically this would be done the other way around checking if !persons.length and returning and then proceeding with the rest of the code. I'm guessing somewhere you will need to handle that this function returns Promise<void> | undefined but we can't just type it as Promise<void> if there is the possibility it could return undefined.